### PR TITLE
Add Prettify API and Improve Build Scripts

### LINKED ISSUE #6286: Include Prettify API to the JSON Module
**Description:**

There is no way to pretty print a JSON value in Ballerina, despite there are many requests in the past. The [most recent request](https://github.com/wso2-enterprise/internal-support-ballerina/issues/632) came from the WSO2 internal apps team.

We have suggested using [`thisarug/prettify`](https://central.ballerina.io/thisarug/prettify/1.0.0) for the users who requested this, but ideally, this should be supported from a `ballerina` library. 

As per the offline discussion, it is decided to include the `thisarug/prettify` library into the `data.jsondata` library.

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose

Fixes: [#6286](https://github.com/ballerina-platform/ballerina-library/issues/6286)
Fixes: https://github.com/wso2-enterprise/internal-support-ballerina/issues/632

## Examples
N/A

## Checklist
- [x] Linked to an issue
- [ ] ~Updated the changelog~
- [x] Added tests
- [ ] ~Updated the spec~
- [x] Checked native-image compatibility

### CODE REVIEW

**Review by hasithaa — DISMISSED** [review]

**Comment by hasithaa** [comment]
@ThisaruGuruge there is a conflict.

**Comment by ThisaruGuruge** [comment]
> @ThisaruGuruge there is a conflict.

Fixed now.

**Comment by prakanth97** [comment]
@ThisaruGuruge you have added `io`, `os` and `file` as dependencies, which means this library belongs to level 4 right? or are these only used for tests. I have to add this to the `ballerina-distribution`.

**Comment by ThisaruGuruge** [comment]
> @ThisaruGuruge you have added `io`, `os` and `file` as dependencies, which means this library belongs to level 4 right? or are these only used for tests. I have to add this to the `ballerina-distribution`.

Even though they are test-only dependencies, we need them to be built after the io, os, and file modules since we create an intermediate pack for testing. The dependency graph resolver will be resolving these accordingly. When adding to the distribution, let's add it under level 4.

File: ballerina/utils.bal [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        return prettifyJsonField(value, indentation, level, isMapField);
+    }
+}
+
+isolated function prettifyJsonMap(map<json> value, string indentation, int level, boolean isMapField) returns string {
+    string initialIndentation = getInitialIndentation(indentation, level, isMapField);
+    string result = string `${initialIndentation}{`;
+    boolean isEmptyMap = value.keys().length() == 0;
+    if !isEmptyMap {
+        result += "\n";
```
Comment (SasinduDilshara) [inline ballerina/utils.bal:34]: Adding `\n` to the string may cause failures in windows, isn't it?

File: ballerina/utils.bal [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        result += indentation;
+    }
+    return result;
+}
+
+isolated function getInitialIndentation(string indentation, int level, boolean isMapField) returns string {
+    if isMapField {
+        return " ";
+    }
+    return getIndentationForLevel(indentation, level);
```
Comment (SasinduDilshara) [inline ballerina/utils.bal]:
```suggestion
    isMapField ? "" : return getIndentationForLevel(indentation, level);
```

File: ballerina/utils.bal (hunk 1 above)
Comment (ThisaruGuruge) [reply ballerina/utils.bal:34]: Windows tests are passing in the PR build. Is there a recommended way to handle this?

**Review by hasithaa — APPROVED** [review]

**Pull request merged**